### DISCUSSION TITLE
How to show/hide hidden files and folders?

### DISCUSSION THREAD

**Discussion by ghost** [SpartanJ/ecode#302] [Q&A] [answered]
`Ctrl+H` doesn't work. I want to edit `.bashrc`.

**Comment by NullPlane** [comment]
For now just add it to your path.
Go `$EDITOR ~/.bashrc` and add an alias for ecode in your directory to call it from shell.
Then just `ecode ~/.bashrc`. If you're not working on a project in a project
directory in-editor filelists are generally not the best-case usage option.

**Reply by ghost** [reply]
I want to open `.bashrc` from ecode. I don't want to use any external shells.

**Comment by SpartanJ** [comment]
You can open the directory and then over the file-tree press right click and "Show hidden files". But usually for any file in the file-system you'll want to just simply configure your file manager to open text files with ecode and that's enough.

**Reply by ghost** [reply]
Where is that button? I don't see anything like you described! Anyway, it's better to have a shortcut key.

**Reply by ghost** [reply]
I don't want to leave the editor. You still have the mindset of Windows and Notepad. Your editor can't be a full featured IDE like VSCode or Vim.

**Comment by SpartanJ** [comment]
Please for questions use the Discussions section [here](https://github.com/SpartanJ/ecode/discussions/categories/q-a).

**Reply by ghost** [reply]
This is not simply a question as it also contains proposal to change or add new features. It's like this, when I asked the question, I'm already sure 80% that the feature is not yet available. Asking is only to confirm and to start a further discussion (aka a feature request).

**Reply by SpartanJ** [reply, marked as answer]
> This is not simply a question as it also contains proposal to change or add new features.

I don't see any proposal. You message was `Ctrl+H doesn't work. I want to edit .bashrc.`.

> I'm already sure 80% that the feature is not yet available. Asking is only to confirm and to start a further discussion (aka a feature request).

I consider that this feature is available.
You have two options:
Simply use "Open File", this option always displays hidden files.

https://github.com/user-attachments/assets/eddae3bf-6c29-4699-8e42-27c989e720ae


Open a Folder and use the "Show hidden files" option.

https://github.com/user-attachments/assets/a7bb273d-714d-401d-a5bc-3955a6fb8630

**Reply by ghost** [reply]
`Open File` is what I used and I don't see any dot files! Yes, I did see dot directories, but no dot files! If it works exactly like in your videos then there must be a bug. The OS I use is MX Linux and I downloaded your Linux binary.

There are some minor annoying issues. If I click into the desktop file it said it's not executable. If I make it executable then it said it can't find ecode. I know that I shouldn't run ecode.bin directly so finally I have to run ecode (the shell script) from the terminal!

**Reply by ghost** [reply]
Perhaps the issue was fixed in newer versions of ecode. I always browse the files using the system's file manager and drag them to open in ecode so I don't use the `Open File` dialog at all.
